src/utils/SafeTransferLib.sol

```diff
@@ -26,34 +31,123 @@ library SafeTransferLib {
         address to,
         uint256 amount
     ) internal {
-        (bool success, bytes memory data) = address(token).call(
-            abi.encodeWithSelector(ERC20.transferFrom.selector, from, to, amount)
-        );
+        bool callStatus;
+
+        assembly {
+            // We'll use 4 + 32 * 3 bytes.
+            let callDataLength := 100
+
+            // Get a pointer to some free memory.
+            let freeMemoryPointer := mload(0x40)
+
+            // Update the free memory pointer for safety.
+            mstore(0x40, add(freeMemoryPointer, callDataLength))
```

Comment:
Do we need to do this? Won't the compiler safely clean before writing to the slot again?